docs/specs/images.md

```diff
@@ -0,0 +1,111 @@
+# Authgear Images
+
+Authgear Images server provides images storing and resizing for user profile attributes.
+
+## Endpoints
+
+### Serving image
+
+The serving image endpoint obtains the images from the object store, performs resizing, and serves the image directly.
+
+The serving image endpoint is publicly accessible.
+
+```
+GET /_images/APP_ID/OBJECT_ID/OPTIONS HTTP/1.1
+```
+
+- `APP_ID`: The authgear app id
+- `OBJECT_ID`: The file object id
+- `OPTIONS`: Following options are supported.
+    - Resizing options, e.g. `w=400,h=300`
+    - Resizing profile, a pre-configured resizing option, e.g. `profile`
+    - Original image: `original`
+
+### Uploading image
+
+The uploading endpoint stores the image to the object store and creates records in the database for future reference.
+
+The uploading image endpoint is a signed URL generated by the Authgear main server and admin server to upload the user profile image.
+
+```
+POST /_images/APP_ID/OBJECT_ID?metadata=METADATA_JSON&signature=SIGNATURE HTTP/1.1
```

Comment:
The metadata should be an opaque string that we do not need to explain what it is. In implementation details it should be `Base64URL(JSON)`